src/browser/modules/D3Visualization/components/OverviewPane.tsx

```diff
@@ -59,17 +89,39 @@ function OverviewPane({
   relationshipCount,
   stats
 }: OverviewPaneProps): JSX.Element {
+  const moreStep = 50
+  const [labelsMax, setLabelsMax] = useState(moreStep)
+  const [relationshipsMax, setRelationshipsMax] = useState(moreStep)
+
+  function onMoreClick(type: any, currentMax: number) {
+    const map: any = {
+      labels: setLabelsMax,
+      relationships: setRelationshipsMax
+    }
+    return (num: number) => map[type](currentMax + num)
+  }
+
   const { relTypes, labels } = stats
+  const visibleLabelKeys = labels ? Object.keys(labels).slice(0, labelsMax) : []
+  const visibleRelationshipKeys = relTypes
+    ? Object.keys(relTypes).slice(0, relationshipsMax)
+    : []
+  const totalNumOfLabelTypes = labels ? Object.keys(labels).length : 0
+  const totalNumOfRelTypes = relTypes ? Object.keys(relTypes).length : 0
 
   return (
     <>
-      <PaneHeader>Overview</PaneHeader>
+      <PaneHeader>{'Overview'}</PaneHeader>
       <PaneBody>
-        {labels && Object.keys(labels).length !== 0 && (
-          <>
-            Node labels
+        {labels && visibleLabelKeys.length !== 0 && (
```

Comment:
with the new visibleLabel keys variable, do we still need to check if "labels" is falsy? I think we can do only `visibleLabelKeys.length !== 0`?

Reply:
I still need it since my IDEA is not smart enough to realize that labels are always defined if visibleLabelKeys.length > 0 on row 132.